### PR TITLE
Border has a 1 pixel thickness even when it's thickness property is set to 0 - fix

### LINKED ISSUE #20156: Border has a 1 pixel thickness even when it's thickness property is set to 0.
### Description

I've got two labels in a vertical stack layout surround by a border so I can round the corners. If you look on the right of the image, the yellow and blue labels have a tiny gap in between which I can't seem to get rid of. The same can be seen between the purple and green.

If I remove the border the gap goes away. Here's the XAML of the right-hand `VerticalStackLayout` with the yellow and blue labels inside.

```
       <VerticalStackLayout Grid.Column="1"
                            Spacing="0"
                            >
           <Border StrokeThickness="0"
                   Margin="5,5,5,0"
                   Padding="0"
                   >
               <Border.StrokeShape>
                   <RoundRectangle CornerRadius="3,3,0,0" />
               </Border.StrokeShape>
               <Label BackgroundColor="{StaticResource Secondary}"
                      HeightRequest="30"
                      HorizontalOptions="Fill" />
           </Border>
           <Border StrokeThickness="0"
                   Margin="5,0,5,5"
                   Padding="0">
               <Border.StrokeShape>
                   <RoundRectangle CornerRadius="0,0,0,0" />
               </Border.StrokeShape>
               <Label BackgroundColor="{StaticResource Tertiary}"
                      HeightRequest="30"
                      HorizontalOptions="Fill" />
               
           </Border>
       </VerticalStackLayout>
```

Surprisingly, if I was to change the `Border StrokeThickness` to any value e.g. `2` and change it back to `0` with hot-reload enabled, the gap will close and look normal as you can see on the right (blue and yellow labels).

Unfortunately as soon as I restart my app or start debugging again and reload, the gap comes back.

### Steps to Reproduce

1- Create a VerticalStackLayout and put a view in there like a Label surrounded by a border.
2- Set the stroke thickness to 0 and notice that there's still a tiny gap in between.

### Link to public reproduction project repository

_No response_

### Version with bug

8.0.3

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 14 API 34

### Did you find any workaround?

No.

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by kubaflo**
### Issues Fixed

Fixes https://github.com/dotnet/maui/issues/20156

|Before|After|
|--|--|
|||

### CODE REVIEW

**Review by pjcollins — DISMISSED** [review]
Fix seems reasonable to me and appears to fix the issue, though I'll defer to @jsuarezruiz for a final review.

File: src/Controls/tests/UITests/Tests/Issues/Issue20156.cs
Comment (jsuarezruiz) [inline src/Controls/tests/UITests/Tests/Issues/Issue20156.cs:21]: Waiting build to generate the snapshots.

**Comment by dustin-wojciechowski** [comment]
Tested to work on IOS, Android, and Windows:
image

**Review by dustin-wojciechowski — DISMISSED** [review]

File: src/Controls/src/Core/Border/Border.cs
Comment (mattleibow) [reply src/Controls/src/Core/Border/Border.cs:311]: I also see the `UpdateStrokeShape` method has a condition for:

```cs
if (StrokeShape is Shape strokeShape && StrokeThickness == 0)
{
    strokeShape.StrokeThickness = StrokeThickness;
}
```

Do we know why we only apply the thickness if it is 0? What happens if the value is say 10?

Comment (mattleibow) [reply src/Controls/src/Core/Border/Border.cs:311]: At the very least, that `0` scares me so a comment is probably worth adding since future someone (aka this Matthew guy) will ask this and maybe do terrible things.

I also wonder if this is worth adding a basic/normal unit test for this? Then there can be also caught there before running all the UI tests: 

 - Core: https://github.com/dotnet/maui/blob/main/src/Core/tests/UnitTests/Views/BorderTests.cs
 - Controls: https://github.com/dotnet/maui/blob/main/src/Controls/tests/Core.UnitTests/BorderUnitTests.cs

I see there like 10 tests total which is surprisingly (I know why so this is a fake surprise) low for this control that is supposedly the future of all containers and borders and other amazingness.

I think a few tests to make sure the thicknesses are propagated correctly to the shape and maybe any other properties that seem to all be intermingled with the shape property.

**Review by mattleibow — COMMENTED** [review]
Thanks for this PR, I feel this issue has been around for some time and was annoying, but now we have fixes.!

I just have a Q on the relationship between a few lines, but I am also hoping to take advantage of your PR and the nice person you are to maybe add some more tests for other permutations of the border, stroke, shape and thickness so that we don't have to write hundreds of UI tests but still protect our future selves from breaking this.

**Comment by kubaflo** [comment]
Hi, @mattleibow you're right I've added a commit with your suggested changes :)
When it comes to this part
```
if (StrokeShape is Shape strokeShape && StrokeThickness == 0)
{
    strokeShape.StrokeThickness = StrokeThickness;
}
```
I don't know why this condition is needed `StrokeThickness == 0`. Everything is okay without it, but I might not have tested it enough. Maybe @jsuarezruiz who introduced this line in this PR https://github.com/dotnet/maui/pull/14403/files might tell us more :)

**Review by mattleibow — DISMISSED** [review]
Waiting on screenshots from CI...

**Pull request merged**
